x-pack/plugins/actions/server/actions_client/actions_client.ts

```diff
@@ -686,9 +686,15 @@ export class ActionsClient {
       AuthorizationMode.RBAC
     ) {
       const additionalPrivileges = this.getSystemActionKibanaPrivileges(actionId, params);
+      // TODO: Optimize so we don't do another get on top of getAuthorizationModeBySource and within the actionExecutor.execute
+      const { attributes } = await this.context.unsecuredSavedObjectsClient.get<RawAction>(
+        'action',
+        actionId
+      );
```

Comment:
I am still testing but probably this addition causes the 404 responses in the functional tests.

An action id is sent to be executed:
```
 const { body: result } = await supertest
        .post(`/api/actions/connector/${ACTION_ID}/_execute`)
```

and this code is trying to fetch it from an SO index but the connector used for the test is preconfigured, so it's in memory not in an SO index.

You can use `this.isPreconfigured(actionId)` to check if it's preconfigured and `this.context.inMemoryConnectors.find` to get it.

Reply:
Thanks @ersin-erdal . let me try to use your suggestion to see if that addresses the problem

Reply:
just re-ran the test. did not try it manually. 

sure. let me re-apply the changes and send it up

Reply:
just FYI: I isolated the test file to only execute one of the failing tests and the code you pointed out is not even being hit (not seeing my `log.info()` entries. so it seems to be happening higher up in the calling tree

Reply:
I just pushed changes that catch that error and just ignore it (logs it only). Lets see how that goes

Reply:
Nice - thanks @ersin-erdal . I was looking at that one yesterday and could not figure it out since it ran successfully locally, but not in CI. Will add the experimental feature flag to that test suite.